src/main/scala/com/wavesplatform/Application.scala

```diff
@@ -99,16 +98,16 @@ class Application(val actorSystem: ActorSystem, val settings: WavesSettings) ext
-  lazy val unconfirmedPoolSynchronizer = actorSystem.actorOf(Props(classOf[UnconfirmedPoolSynchronizer], newTransactionHandler, settings.utxSettings, networkController, utxStorage))
-  lazy val coordinator = actorSystem.actorOf(Props(classOf[Coordinator], this), "Coordinator")
-  lazy val scoreObserver = actorSystem.actorOf(Props(classOf[ScoreObserver], networkController, coordinator, settings.synchronizationSettings), "ScoreObserver")
-  lazy val historyReplier = actorSystem.actorOf(Props(classOf[HistoryReplier], networkController, history: History, settings.synchronizationSettings.maxChainLength: Int), "HistoryReplier")
-  lazy val blockGenerator = actorSystem.actorOf(Props(classOf[BlockGeneratorController], settings.minerSettings, history, time, peerManager,
-    wallet, stateReader, settings.blockchainSettings, utxStorage, coordinator), "BlockGenerator")
-  lazy val blockchainSynchronizer = actorSystem.actorOf(Props(classOf[BlockchainSynchronizer], networkController, coordinator, history, settings.synchronizationSettings), "BlockchainSynchronizer")
-  lazy val peerSynchronizer = actorSystem.actorOf(Props(classOf[PeerSynchronizer], this), "PeerSynchronizer")
+  lazy val networkController: ActorRef = actorSystem.actorOf(Props(new NetworkController(settings.networkSettings, upnp, peerManager, messagesHandler)), "NetworkController")
+  lazy val peerManager: ActorRef = actorSystem.actorOf(PeerManager.props(settings.networkSettings, networkController, settings.blockchainSettings.addressSchemeCharacter), "PeerManager")
+  lazy val unconfirmedPoolSynchronizer: ActorRef = actorSystem.actorOf(Props(new UnconfirmedPoolSynchronizer(newTransactionHandler, settings.utxSettings, networkController, utxStorage)))
+  lazy val coordinator: ActorRef = actorSystem.actorOf(Props(new Coordinator(networkController, blockchainSynchronizer, blockGenerator, peerManager, scoreObserver, blockchainUpdater, time, utxStorage, history, stateReader, checkpoints, settings)), "Coordinator")
+  lazy val scoreObserver: ActorRef = actorSystem.actorOf(Props(new ScoreObserver(networkController, coordinator, settings.synchronizationSettings)), "ScoreObserver")
+  lazy val historyReplier: ActorRef = actorSystem.actorOf(Props(new HistoryReplier(networkController, history: History, settings.synchronizationSettings.maxChainLength: Int)), "HistoryReplier")
+  lazy val blockGenerator: ActorRef = actorSystem.actorOf(Props(new BlockGeneratorController(settings.minerSettings, history, time, peerManager,
+    wallet, stateReader, settings.blockchainSettings, utxStorage, coordinator)), "BlockGenerator")
+  lazy val blockchainSynchronizer: ActorRef = actorSystem.actorOf(Props(new BlockchainSynchronizer(networkController, coordinator, history, settings.synchronizationSettings)), "BlockchainSynchronizer")
+  lazy val peerSynchronizer: ActorRef = actorSystem.actorOf(Props(new PeerSynchronizer(networkController, peerManager, settings.networkSettings)), "PeerSynchronizer")
```

Comment:
Neither of these actors is used outside `Application`. I suggest moving all of these values inside the `run()` method and getting rid of 'de-lazying'.